### ISSUE TITLE
Issue in Function Type

### ISSUE THREAD

**Issue by srehanuddin** [Microsoft/TypeScript#3585]
```ts
function callback(){
	return "My String";
}	
function myFunction(str : string, cb : (mystr : string)=>string){
}
myFunction("Hello", callback)
```
As per my knowledge this mentioned code should show error on compilation time because myFunction accept two parameters, 1st is string type and 2nd is function which accept 1 parameter of string, but i passed the function which have no arguments.

**Comment by RyanCavanaugh** [comment]
See https://github.com/Microsoft/TypeScript/wiki/Type%20Compatibility#comparing-two-functions
